Merge/foundation master 20200714

meowsbits opened the pull request:
Merges the last few commits unique to EF master prior to their v1.9.16 release, as well as a few subsequent commits since.

soc1c commented:
restarted travis

soc1c commented:
all green: https://travis-ci.org/etclabscore/multi-geth/builds/649849608?utm_source=github_status&utm_medium=notification

meowsbits commented:
Releasing this merged version to the testnets, some of which have already "begun" running EIP2200, posing a consistency problem.

It could break the testnets.

soc1c commented:
Testnets are already broken and need to be rolled back. All we need is releases for Multi Geth and Parity.

fjl committed:
common/math: use math/bits intrinsics for Safe* (#21316)

This is a resubmit of ledgerwatch/turbo-geth#556. The performance
benefit of this change is negligible, but it does remove a TODO.

renaynay committed:
cmd/geth, cmd/puppeth: replace deprecated rpc and ws flags in tests and docs (#21317)

rjl493456442 committed:
accounts/external: remove dependency on internal/ethapi (#21319)

Fixes #20535

Co-authored-by: Felix Lange <fjl@twurst.com>

fjl committed:
params: go-ethereum v1.9.16 stable

fjl committed:
params: begin v1.9.17 release cycle

rjl493456442 committed:
les: historical data garbage collection (#19570)

This change introduces garbage collection for the light client. Historical
chain data is deleted periodically. If you want to disable the GC, use
the --light.nopruning flag.

fjl committed:
light: goimports -w (#21325)

libotony committed:
p2p/discv5: unset pingEcho on pong timeout (#21324)

tiendo1011 committed:
README.md: point Go API reference link to pkg.go.dev (#21321)

rjl493456442 committed:
internal/ethapi: cap txfee for SignTransaction and Resend (#21231)

renaynay committed:
.github: Change Code Owners (#21326)

* modify code owners

* add marius

rjl493456442 committed:
core/rawdb: better log messages for ancient failure (#21327)

fjl committed:
p2p/discover: require table nodes to have an IP (#21330)

This fixes a corner case in discv5. The issue cannot happen in discv4
because it performs IP checks on all incoming node information.

renaynay committed:
cmd/clef: change --rpcport to --http.port and update flags in docs (#21318)

holiman committed:
core: transaction pool optimizations (#21328)

* core: added local tx pool test case

* core, crypto: various allocation savings regarding tx handling

* core/txlist, txpool: save a reheap operation, avoid some bigint allocs

Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>

meowsbits committed:
Merge remote-tracking branch 'foundation/master' into merge/foundation-master-20200714

Conflicts:
      .github/CODEOWNERS
      README.md
      cmd/geth/main.go
      cmd/geth/usage.go
      consensus/clique/clique.go
      core/rawdb/database.go
      core/rawdb/freezer.go
      eth/downloader/downloader.go
      internal/ethapi/api.go
      les/client.go
      les/server.go
      params/version.go

meowsbits committed:
rawdb,ethapi: fix minor straggling merge conflicts

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
main,downloader: (lint) goimports

Signed-off-by: meows <b5c6@protonmail.com>
